src/validations/prefab/component.ts

```diff
@@ -92,18 +94,36 @@ const componentSchema = (
   const canValidateStyle = styleType && styleValidator[styleType];
 
   return Joi.object({
-    name: Joi.string().required(),
+    name: Joi.string().when('type', {
```

Comment:
I'm not a Joi expert, but can't we do the `is: 'COMPONENT'` check on the `Joi.object` level? Instead of doing every key separate. I think it will be a bit more readable.

Reply:
Thank for the good suggestion, we have split up the component schema and partial schema